com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -691,10 +691,41 @@ void UpdateRewardStats()
         /// <summary>
         /// Sets the done flag to true and resets the agent.
         /// </summary>
+        /// <remarks>
+        /// This should be used when the episode can no longer continue, such as when the Agent
+        /// reaches the goal or fails at the task.
+        /// </remarks>
         /// <seealso cref="OnEpisodeBegin"/>
+        /// <seealso cref="EpisodeMaxStepReached"/>
         public void EndEpisode()
         {
-            NotifyAgentDone(DoneReason.DoneCalled);
+            EndEpisodeAndReset(DoneReason.DoneCalled);
+        }
+
+        /// <summary>
+        /// Indicate that the episode has reached a maximum number of steps.
+        /// This has the same end result as calling <see cref="EndEpisode"/>, but has a
+        /// slightly different effect on training.
+        /// </summary>
+        /// <remarks>
+        /// This should be used when the episode could continue, but has gone on for
+        /// a sufficient number of steps.
+        /// </remarks>
+        /// <seealso cref="OnEpisodeBegin"/>
+        /// <seealso cref="EndEpisode"/>
+        public void EpisodeMaxStepReached()
```

Comment:
I would prefer not calling this MaxStepReached, but call it Interrupted. The reason is that MaxStepReached will not (I hope) be the only reason an agent will be interrupted forever. A user triggered scene reset should have the same effect as MaxStepReached (for example if the Agents reset but not because it is their fault)

Reply:
ok, so just rename this to `EpisodeInterrupted` (and change the docs accordingly)?